meta-facebook/at-cb/src/platform/plat_hook.c

```diff
@@ -771,6 +771,14 @@ bool pre_pex89000_read(sensor_cfg *cfg, void *args)
 	mux_config *pre_args = (mux_config *)args;
 	pre_args->bus = cfg->port;
 
+	/* Check if switch is ready */
+	if (get_board_revision() > EVT2_STAGE) {
+		if (is_sw_ready(cfg->num) != true) {
```

Comment:
I suppose the switch ready status is being set in a different thread. Are we sure this won't get read before the other thread has got the status update via sw_heartbeat_read? Is there any synchronization between the two? Is there any retry logic in terms of callers for is_sw_ready() in case it fails initially?

Reply:
Hi @jagpalgill 
Modify with your suggestions, we add mutex for synchronization . please help to review this commit again, thank you.

Reply:
Mutex will solve the problem where reader & writer are trying to access the ready state at same time. My question was more on situation where reader reads the value before writer even sets it because the read & write is happening in a different thread. Would we have that situation? If yes, are we fine with that situation, for example because we have a retry logic in place? If no, then how are we planning on handling it?

Reply:
@jagpalgill ,
When BIC is boot-up, the switch ready cache defaults to false. We use a thread to monitor switch heartbeat to confirm switch ready after DC on, so reader will reads the value after writer sets it.
The switch temperature can't be read at the beginning of DC on is expected.

Regards,
Lora.